x-pack/plugin/eql/qa/common/src/main/java/org/elasticsearch/test/eql/CommonEqlActionTestCase.java

```diff
@@ -147,12 +148,13 @@ public void cleanup() throws Exception {
     public static List<Object[]> asArray(List<EqlSpec> specs) {
         AtomicInteger counter = new AtomicInteger();
         return specs.stream().map(spec -> {
-            String name = spec.description();
-            if (Strings.isNullOrEmpty(name)) {
-                name = spec.note();
+            String name = spec.query();
+
+            if (Strings.isNullOrEmpty(spec.description()) == false) {
+                name = spec.description();
             }
-            if (Strings.isNullOrEmpty(name)) {
-                name = "" + (counter.get() + 1);
+            else if (Strings.isNullOrEmpty(spec.note()) == false) {
```

Comment:
Unless all tests have a note, please keep the name counter since otherwise the name becomes the query which makes it impossible to rerun the test (since it's not a correct method name).